### PR TITLE
docs: Docker driver supports task user option

### LINKED ISSUE #6519: task `user` should set the user for the docker driver
Docker supports setting the user for a contained task on the command line, but nomad does not use the task `user` key to do so.

### PR DESCRIPTION
**Pull request by notnoop**
Also, add a test case.

Closes #6519

### CODE REVIEW

File: website/source/docs/drivers/docker.html.md [hunk 1]
```diff
@@ -413,6 +413,8 @@ The `docker` driver supports the following configuration in the job spec.  Only
 * `pids_limit` - (Optional) An integer value that specifies the pid limit for
   the container. Defaults to unlimited.
 
+Additionally, docker driver supports customization of the container user through task [`user` option](/docs/job-specification/task.html#user).
```
Comment (endocrimes) [inline website/source/docs/drivers/docker.html.md]:
```suggestion
Additionally the docker driver supports customization of the container's user through the task [`user` option](/docs/job-specification/task.html#user).
```

grammar nit

Comment (jippi) [reply website/source/docs/drivers/docker.html.md]: Grammarly would insist on adding the `,`  :) it always fights me on that

Comment (shantanugadgil) [reply website/source/docs/drivers/docker.html.md]: > 
> 
> Grammarly would insist on adding the `,` :) it always fights me on that

the comma after "additionally" should be there, and the "the" is also correct.
while we are grammar nitting 😁 I usually avoid using apostrophes. So, I might reqrite the sentence as:

>Additionally, the docker driver supports customization of the container's user, through the [`user` option](/docs/job-specification/task.html#user) of the task.

🤷‍♂

Comment (notnoop) [reply website/source/docs/drivers/docker.html.md]: Um.. why is a comma not warranted here?  I learned it with a comma, and I see many online references advocating the same:

Lexico (powered by Oxford) has it in the example 
> additionally, 50 hours of practical experience will be necessary
https://www.lexico.com/en/definition/additionally

Wheaten College Writing Resources lists Additionally and points need for a comma
> Many adverbs that end in –ly and transitions at the beginning of a sentence need to be followed by a comma, too.
https://www.wheaton.edu/academics/services/writing-center/writing-resources/the-comma/

In somewhat related StackExchange answer about additionally and other words:
> If starting a sentence with an introductory word or phrase then, yes, a comma would be required.
https://english.stackexchange.com/questions/21532/when-to-use-commas-in-a-sentence-that-starts-with-finally-additionally-etc

Comment (notnoop) [reply website/source/docs/drivers/docker.html.md]: Interestingly the nomad repo contains 42 instances of `Additionally,` and 7 cases of `Additionally ` (without comma):

```
$ git grep  'Additionally ' |wc -l
       7
$ git grep  'Additionally,' |wc -l
      42
```

Comment (endocrimes) [reply website/source/docs/drivers/docker.html.md]: dropping the comma was accidental :joy:

**Pull request merged**